ldaptor/test/test_ldiftree.py

```diff
@@ -6,14 +6,25 @@
 import random
 import errno
 import shutil
+import functools
 
+from twisted.python import log
 from twisted.trial import unittest
+from twisted.internet import defer, testing
 
 from ldaptor import ldiftree, entry, delta, testutil
 from ldaptor.entry import BaseLDAPEntry
 from ldaptor.protocols.ldap import ldaperrors, ldifprotocol
 
 
+def _fromCoroutineFunction(corofn):
```

Comment:
I think that this can be public (don't start with underscore) and have a docstring.

When the deferred already has the result and there is no need to spin the reactor, you can use `successResultOf` https://twistedmatrix.com/documents/current/api/twisted.trial._synctest._Assertions.html#successResultOf

Reply:
a coroutine that never yields won't spin the reactor

Reply:
I've made this public, and put it into ldaptor.test.util